nifi-registry/nifi-registry-core/nifi-registry-web-api/src/main/java/org/apache/nifi/registry/web/api/BucketFlowResource.java

```diff
@@ -295,6 +295,48 @@ public Response createFlowVersion(
         return Response.status(Response.Status.OK).entity(createdSnapshot).build();
     }
 
+    @POST
+    @Path("{flowId}/versions/migrate")
```

Comment:
Having a verb `migrate` in the path does not follow general REST API conventions, where the path should be a noun and the HTTP method is the verb. One option could be to make `migrate` and `Boolean` query parameter on the `/versions` resource method. However, the word `migrate` itself does not clearly convey what is happening, so perhaps another parameter name would be better. As the method name mentions keeping the author, it seems like a better approach would be to adding a query parameter named something like `preserveAuthor`. If there are other properties to be preserved, perhaps it could be named `preserveSourceProperties`.

Reply:
We had discussed this offline and landed on the `/migrate` approach to clearly separate the usage of the API. The idea was that no one should really be calling the `/migrate` end-point, except for a specific utility that wants to move data between two registries. 

However, I agree that it generally does not follow REST standards, so I'm not that opposed to using a query parameter on the existing end-points. If we were to go with a query parameter then it should probably be `preserveSourceProperties` because it is more general than just the `author`. 

We have 3 different entities that would be migrated:

- `Buckets` - need to preserve id, rather than generating new
- `Flows` - need to preserve id, rather than generating new
- `Flow Versions` - need to preserve id and author, rather than generating new and using authenticated user

Reply:
Thanks for the reply and background @bbende.

Unfortunately JAX-RS does not appear to support method-based matching using Request query parameters, but it seems like introducing a `preserveSourceProperties` parameter in the existing method would provide a clear separation of behavior.